internal/pkg/manifest/env.go

```diff
@@ -95,9 +95,47 @@ type environmentNetworkConfig struct {
 }
 
 type environmentVPCConfig struct {
-	ID      *string              `yaml:"id,omitempty"`
-	CIDR    *IPNet               `yaml:"cidr,omitempty"`
-	Subnets subnetsConfiguration `yaml:"subnets,omitempty"`
+	ID                  *string              `yaml:"id,omitempty"`
+	CIDR                *IPNet               `yaml:"cidr,omitempty"`
+	Subnets             subnetsConfiguration `yaml:"subnets,omitempty"`
+	SecurityGroupConfig securityGroupConfig  `yaml:"security_group,omitempty"`
+}
+
+type securityGroupConfig struct {
+	Ingress []securityGroupRules `yaml:"ingress,omitempty"`
+	Egress  []securityGroupRules `yaml:"egress,omitempty"`
+}
+
+// IsEmpty returns true if there is no security group rule defined.
+func (sgc securityGroupConfig) IsEmpty() bool {
+	return len(sgc.Ingress) == 0 && len(sgc.Egress) == 0
+}
+
+type securityGroupRules struct {
+	CidrIp     string `yaml:"cidr"`
```

Comment:
nit: https://github.com/golang/go/wiki/CodeReviewComments#initialisms
```suggestion
	CidrIP     string `yaml:"cidr"`
```